src/NuGet.Core/NuGet.PackageManagement/IDE/PackageRestoreManager.cs

```diff
@@ -168,17 +168,24 @@ public void EnableCurrentSolutionForRestore(bool fromActivation)
                     continue;
                 }
 
-                var nuGetProjectName = NuGetProject.GetUniqueNameOrName(nuGetProject);
-                var installedPackageReferences = await nuGetProject.GetInstalledPackagesAsync(token);
-                foreach (var installedPackageReference in installedPackageReferences)
+                try
                 {
-                    List<string> projectNames = null;
-                    if (!packageReferencesDict.TryGetValue(installedPackageReference, out projectNames))
+                    var nuGetProjectName = NuGetProject.GetUniqueNameOrName(nuGetProject);
+                    var installedPackageReferences = await nuGetProject.GetInstalledPackagesAsync(token);
+                    foreach (var installedPackageReference in installedPackageReferences)
                     {
-                        projectNames = new List<string>();
-                        packageReferencesDict.Add(installedPackageReference, projectNames);
+                        List<string> projectNames = null;
+                        if (!packageReferencesDict.TryGetValue(installedPackageReference, out projectNames))
+                        {
+                            projectNames = new List<string>();
+                            packageReferencesDict.Add(installedPackageReference, projectNames);
+                        }
+                        projectNames.Add(nuGetProjectName);
                     }
-                    projectNames.Add(nuGetProjectName);
+                }
+                catch (Exception)
```

Comment:
not sure if this is a good idea to catch every exception here. can we be more specific of what exceptions we are expecting?